doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,388 @@
+
+# Preparing networks to be inferenced with G-API {#gapi_fb_prep_nets}
+
+Let's prepare our chosen networks for being processed inside the graph.
+Providing a new network support is quite similar to a new operation
+implementation.
+
+To declare a deep learning topologies we should use the specific macro
+`G_API_NET()`:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp net_decl
+
+Similarly to `G_API_OP()` macro for a new operation, it takes three arguments to
+register a new type.
+They are:
+1. Network interface name --- also serves as a name of a new type defined
+   with this macro;
+2. Network signature      --- an `std::function<>`-like signature which defines
+   inputs and outputs;
+3. Network's unique name  --- used to identify it within the system when the
+   type is stripped.
+
+# Graph compilation {#gapi_fb_comp}
+
+For further convenience we should use the version of a pipeline expression
+with a lambda-based constructor to keep all temporary objects in a dedicated
+scope (almost all the graph's body is skipped in this code snippet
+except for input and output nodes):
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp comp_str_1
+...
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp comp_str_2
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp comp_str_3
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp comp_str_4
+
+To apply a model inside the graph, we should call
+`cv::gapi::infer<>()` setting a network type as a template argument to
+specify the topology we want to infer:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp net_usg_fd
+
+If we need to convey several ROIs for inference, we can use
+the `cv::gapi::infer<>()` function's overload:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp net_usg_ld
+
+The second argument of it is the image to be processed, and the first is a
+`cv::GArray<>` of ROIs; the network will be automatically inferred for every single ROI.
+
+# Custom operations implementation {#gapi_fb_proc}
+
+A new kernel implementation has been already described clearly in OpenCV G-API
+documentation (see [G-API Kernel API](@ref gapi_kernel_api)).
+
+We usually use `G_TYPED_KERNEL()` macro to define a new kernel type; a small yet
+significant adjustment has to be made when  defining a kernel with a
+multiple return value. For this case, we must use `G_TYPED_KERNEL_M()` macro:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp kern_m_decl
+
+Note: the `std::tuple` appearance in a kernel signature could
+be the sign to change the macro. At the implementation step, all the output
+arguments of a kernel signature are just put after input ones keeping the order
+if they are multiple:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp kern_m_impl
```

Comment:
Just drop this section

Reply:
Stopped here. Everything above can be addressed already. Will review the rest till EOD.